compiler/packages/babel-plugin-react-compiler/src/TypeInference/InferTypes.ts

```diff
@@ -483,3 +521,9 @@ class Unifier {
     return type;
   }
 }
+
+const RefLikeNameRE = /^(?:[a-zA-Z$_][a-zA-Z$_0-9]*)Ref$|ref$/;
```

Comment:
I think you need to update the grouping or add `^` before 'ref$', because this will match things like `abcref`. The `^` is only applying to the "...Ref" part of the or.

Reply:
Ah, good catch. Done